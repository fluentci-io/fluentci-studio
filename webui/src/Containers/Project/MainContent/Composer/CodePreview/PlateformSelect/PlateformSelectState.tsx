import { atom } from "recoil";

export type PlateformItem = { id: Plateform; label: string; filename: string };
export type Plateform = "github" | "gitlab" | "aws" | "circleci" | "azure";

export const PlateformSelectState = atom<PlateformItem[]>({
  key: "PlateformSelectState",
  default: [
    {
      id: "github",
      label: "Github Actions",
      filename: "ci.yml",
    },
  ],
});
