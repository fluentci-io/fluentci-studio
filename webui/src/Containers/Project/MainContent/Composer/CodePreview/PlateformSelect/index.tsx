import PlateformSelect from "./PlateformSelectWithData";

export default PlateformSelect;
