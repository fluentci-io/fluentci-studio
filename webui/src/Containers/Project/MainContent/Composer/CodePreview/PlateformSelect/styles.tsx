import styled from "@emotion/styled";

export const Container = styled.div`
  width: 250px;
  margin-top: 20px;
`;

export default {
  Select: {
    Root: {
      style: {
        fontFamily: "Lexend",
      },
    },
    DropdownListItem: {
      style: {
        fontFamily: "Lexend",
      },
    },
    ControlContainer: {
      style: {
        border: "1px solid #ffffff45",
      },
    },
  },
};
