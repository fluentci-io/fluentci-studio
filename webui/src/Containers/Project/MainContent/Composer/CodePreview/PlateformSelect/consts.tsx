import { PlateformItem } from "./PlateformSelectState";

export const plateforms: PlateformItem[] = [
  {
    label: "Azure Pipelines",
    id: "azure",
    filename: "azure-pipelines.yml",
  },
  {
    label: "AWS CodePipeline",
    id: "aws",
    filename: "buildspec.yml",
  },
  {
    label: "Circle CI",
    id: "circleci",
    filename: "config.yml",
  },
  {
    label: "Github Actions",
    id: "github",
    filename: "ci.yml",
  },
  {
    label: "GitLab CI",
    id: "gitlab",
    filename: ".gitlab-ci.yml",
  },
];
