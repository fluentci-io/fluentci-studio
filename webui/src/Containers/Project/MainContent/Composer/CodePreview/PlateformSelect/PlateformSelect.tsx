import { FC } from "react";
import { PlateformItem } from "./PlateformSelectState";
import { Select } from "baseui/select";
import styles, { Container } from "./styles";

export type PlateformSelectProps = {
  current: PlateformItem[];
  onSelect: (plateform: PlateformItem[]) => void;
  plateforms: PlateformItem[];
};

const PlateformSelect: FC<PlateformSelectProps> = (props) => {
  const { current, plateforms, onSelect } = props;
  return (
    <Container>
      <Select
        overrides={styles.Select}
        clearable={false}
        options={plateforms}
        value={current}
        placeholder="Select color"
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        onChange={(params: { value: any }) => {
          onSelect(params.value);
        }}
      />
    </Container>
  );
};

export default PlateformSelect;
