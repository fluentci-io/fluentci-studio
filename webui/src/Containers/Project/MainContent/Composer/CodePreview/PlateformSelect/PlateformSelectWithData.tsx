import { FC } from "react";
import PlateformSelect from "./PlateformSelect";
import { plateforms } from "./consts";
import { useRecoilState } from "recoil";
import { PlateformSelectState } from "./PlateformSelectState";

const PlateformSelectWithData: FC = () => {
  const [currentPlateform, setCurrentPlateform] =
    useRecoilState(PlateformSelectState);

  return (
    <PlateformSelect
      current={currentPlateform}
      plateforms={plateforms}
      onSelect={setCurrentPlateform}
    />
  );
};

export default PlateformSelectWithData;
