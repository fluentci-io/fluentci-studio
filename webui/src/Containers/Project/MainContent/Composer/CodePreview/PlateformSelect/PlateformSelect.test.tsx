import { render } from "@testing-library/react";
import PlateformSelect from "./PlateformSelect";

describe("PlateformSelect", () => {
  it("should render the component", () => {
    const { container } = render(
      <PlateformSelect
        current={[
          {
            id: "github",
            label: "Github Actions",
            filename: "ci.yml",
          },
        ]}
        plateforms={[
          {
            filename: "azure-pipelines.yml",
            id: "azure",
            label: "Azure Pipelines",
          },
          { filename: "buildspec.yml", id: "aws", label: "AWS CodePiepline" },
          { filename: "config.yml", id: "circleci", label: "CirecleCI" },
          { filename: "ci.yml", id: "github", label: "Github Actions" },
          { filename: ".gitlab-ci.yml", id: "gitlab", label: "Gitlab CI" },
        ]}
        onSelect={vi.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
