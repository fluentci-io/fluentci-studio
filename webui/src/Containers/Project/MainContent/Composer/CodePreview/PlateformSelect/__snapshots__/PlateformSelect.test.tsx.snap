// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlateformSelect > should render the component 1`] = `
<div>
  <div
    class="css-38orqy"
  >
    <div
      class=""
      data-baseweb="select"
    >
      <div
        class=""
      >
        <div
          class=""
        >
          <div
            class=""
            value="github"
          >
            Github Actions
          </div>
          <div
            class=""
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="Selected Github Actions. "
              class=""
              role="combobox"
              tabindex="0"
              value=""
            />
            <div
              class=""
            />
          </div>
        </div>
        <div
          class=""
        >
          <svg
            class=""
            data-baseweb="icon"
            title="open"
            viewBox="0 0 24 24"
          >
            <title>
              open
            </title>
            <path
              d="M12.7071 15.2929L17.1464 10.8536C17.4614 10.5386 17.2383 10 16.7929 10L7.20711 10C6.76165 10 6.53857 10.5386 6.85355 10.8536L11.2929 15.2929C11.6834 15.6834 12.3166 15.6834 12.7071 15.2929Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
